### PR TITLE
feat: add config file

### PR DESCRIPTION
**Pull request by reliveyy**
This commit adds a config file (toml format) to xud-docker. A new container named "exchangeunion/utils" is added to help setup.sh parsing command line arguments and the config file.

### CODE REVIEW

**Comment by kilrau** [comment]
Still:
```
kilrau@K-Yoga:~$ xud -b feat/config-file
1) Simnet
2) Testnet
3) Mainnet
Please choose the network: 3
Unable to find image 'exchangeunion/utils:latest' locally
docker: Error response from daemon: pull access denied for exchangeunion/utils, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
/dev/fd/63: line 532: CONFIG: unbound variable
```

**Comment by kilrau** [comment]
Image is being downloaded now, but still failing:
```
kilrau@K-Yoga:~/docs$ xud -b feat/config-file
1) Simnet
2) Testnet
3) Mainnet
Please choose the network: 2
Unable to find image 'exchangeunion/utils:latest' locally
latest: Pulling from exchangeunion/utils
89d9c30c1d48: Already exists
910c49c00810: Pulling fs layer
66564da92047: Pulling fs layer
bb0265c0d9a9: Pulling fs layer
b86dfbfbb035: Pulling fs layer
32a49b5c84b3: Pulling fs layer
743c91f5fc67: Pulling fs layer
a62fc58c0f48: Pulling fs layer
32a49b5c84b3: Waiting
743c91f5fc67: Waiting
a62fc58c0f48: Waiting
b86dfbfbb035: Waiting
bb0265c0d9a9: Verifying Checksum
bb0265c0d9a9: Download complete
910c49c00810: Download complete
910c49c00810: Pull complete
32a49b5c84b3: Verifying Checksum
32a49b5c84b3: Download complete
b86dfbfbb035: Verifying Checksum
b86dfbfbb035: Download complete
743c91f5fc67: Verifying Checksum
743c91f5fc67: Download complete
a62fc58c0f48: Download complete
66564da92047: Verifying Checksum
66564da92047: Download complete
66564da92047: Pull complete
bb0265c0d9a9: Pull complete
b86dfbfbb035: Pull complete
32a49b5c84b3: Pull complete
743c91f5fc67: Pull complete
a62fc58c0f48: Pull complete
Digest: sha256:fea06c2ad4d03f41435e25e5d87c0844af0262f33da527be0a3edde59c2b28a4
Status: Downloaded newer image for exchangeunion/utils:latest
/dev/fd/63: line 515: PROJECT_DIR: unbound variable
```

**Comment by reliveyy** [comment]
New error on Arch Linux: failed to add the host (vethd65308b) <=> sandbox (veth9f149e4) pair interfaces: operation not supported.
/dev/fd/63: line 532: CONFIG: unbound variable

<details>
<summary>more details</summary>
<pre>
[yy@pc ~]$ ~/xud.sh -b feat/config-file
1) Simnet
2) Testnet
3) Mainnet
Please choose the network: 2 
Unable to find image 'exchangeunion/utils:latest' locally
latest: Pulling from exchangeunion/utils
89d9c30c1d48: Already exists
910c49c00810: Pulling fs layer
66564da92047: Pulling fs layer
bb0265c0d9a9: Pulling fs layer
b86dfbfbb035: Pulling fs layer
32a49b5c84b3: Pulling fs layer
743c91f5fc67: Pulling fs layer
a62fc58c0f48: Pulling fs layer
743c91f5fc67: Waiting
b86dfbfbb035: Waiting
a62fc58c0f48: Waiting
bb0265c0d9a9: Verifying Checksum
bb0265c0d9a9: Download complete
910c49c00810: Verifying Checksum
910c49c00810: Download complete
910c49c00810: Pull complete
66564da92047: Verifying Checksum
66564da92047: Download complete
b86dfbfbb035: Verifying Checksum
b86dfbfbb035: Download complete
32a49b5c84b3: Verifying Checksum
32a49b5c84b3: Download complete
66564da92047: Pull complete
bb0265c0d9a9: Pull complete
743c91f5fc67: Verifying Checksum
743c91f5fc67: Download complete
a62fc58c0f48: Verifying Checksum
a62fc58c0f48: Download complete
b86dfbfbb035: Pull complete
32a49b5c84b3: Pull complete
743c91f5fc67: Pull complete
a62fc58c0f48: Pull complete
Digest: sha256:fea06c2ad4d03f41435e25e5d87c0844af0262f33da527be0a3edde59c2b28a4
Status: Downloaded newer image for exchangeunion/utils:latest
docker: Error response from daemon: failed to create endpoint silly_morse on network bridge: failed to add the host (vethd65308b) <=> sandbox (veth9f149e4) pair interfaces: operation not supported.
/dev/fd/63: line 532: CONFIG: unbound variable
</pre>

**Comment by reliveyy** [comment]
> New error on Arch Linux: failed to add the host (vethd65308b) <=> sandbox (veth9f149e4) pair interfaces: operation not supported.

See https://serverfault.com/questions/738773/docker-failed-to-add-the-pair-interfaces-operation-not-supported. So probably not related to this PR. Reboot the system should fix this issue.

**Comment by kilrau** [comment]
Needs a rebase (e.g. still using geth 1.9.7) @reliveyy

File: images/utils/config-parser.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    "HOME_DIR": "$HOME/.xud-docker",
+    "NETWORK_DIR": "$HOME_DIR/$NETWORK",
+    "PROJECT_DIR": None,
+    "BITCOIND_DIR": "$NETWORK_DIR/data/bitcoind",
+    "LITECOIND_DIR": "$NETWORK_DIR/data/litecoind",
+    "GETH_DIR": "$NETWORK_DIR/data/geth",
+    "GETH_CHAINDATA_DIR": "$NETWORK_DIR/data/geth/chaindata",
+    "LOGFILE": "$NETWORK_DIR/xud-docker.log",
+    "DISABLE_UPDATES": "false",
+    "CONFIG":  "$HOME_DIR/xud-docker.conf",
```
Comment (kilrau) [inline images/utils/config-parser.py]: I started my existing testnet environment with `xud -b feat/config-file`, but can't find any xud-docker.conf file. That the desired behavior?

Comment (kilrau) [reply images/utils/config-parser.py]: Maybe because it's not initialized with any values, so it doesn't exist?

Comment (reliveyy) [reply images/utils/config-parser.py]: If you do not provide your xud-docker.conf then default configuration options will be applied and no xud-docker.conf file will be created by the script. You can copy the `sample-xud-docker.conf` from this project to customize some options.

Comment (kilrau) [reply images/utils/config-parser.py]: https://github.com/ExchangeUnion/xud-docker/pull/233#pullrequestreview-326089665 will solve this UX problem.

**Comment by kilrau** [comment]
```
kilrau@K-Yoga:~$ xud -b feat/config-file
1) Simnet
2) Testnet
3) Mainnet
Please choose the network: 2
/dev/fd/63: line 537: CONFIG_FILE: unbound variable
```

**Pull request closed without merging**